src/voice/audio.rs

```diff
@@ -51,26 +96,53 @@ impl Audio {
             volume: 1.0,
             finished: false,
             source,
+            position: Duration::new(0, 0),
+            position_modified: false,
         }
     }
 
+    /// Sets [`playing`] to `true` in a manner that allows method chaining.
+    ///
+    /// [`playing`]: #structfield.playing
     pub fn play(&mut self) -> &mut Self {
         self.playing = true;
 
         self
     }
 
+    /// Sets [`playing`] to `false` in a manner that allows method chaining.
+    ///
+    /// [`playing`]: #structfield.playing
     pub fn pause(&mut self) -> &mut Self {
         self.playing = false;
 
         self
     }
 
+    /// Sets [`volume`] in a manner that allows method chaining.
+    ///
+    /// [`volume`]: #structfield.volume
     pub fn volume(&mut self, volume: f32) -> &mut Self {
         self.volume = volume;
 
         self
     }
+
+    /// Change the in the stream for subsequent playback.
+    /// Currently a No-op.
+    pub fn position(&mut self, position: Duration) -> &mut Self {
+        self.position = position;
+        self.position_modified = true;
+
+        self
+    }
+
+    /// Steps playback location forward by one frame.
+    /// *Used internally.*
+    pub fn step_frame(&mut self) {
```

Comment:
If something is only intended to be used internally, it should be marked with a visibility of `pub(crate)`. Otherwise, some comment about how "this is normally meant for internal use, but you may find it to be useful for X or Y scenarios" is good.